modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPConfigurationListLocalServiceImpl.java

```diff
@@ -19,4 +39,178 @@
+
+		cpConfigurationList.setExternalReferenceCode(externalReferenceCode);
+		cpConfigurationList.setGroupId(groupId);
+		cpConfigurationList.setCompanyId(user.getCompanyId());
+		cpConfigurationList.setUserId(user.getUserId());
+		cpConfigurationList.setUserName(user.getFullName());
+		cpConfigurationList.setParentCPConfigurationListId(
+			parentCPConfigurationListId);
+		cpConfigurationList.setMasterCPConfigurationList(
+			masterCPConfigurationList);
+		cpConfigurationList.setName(name);
+		cpConfigurationList.setPriority(priority);
+		cpConfigurationList.setDisplayDate(displayDate);
+		cpConfigurationList.setExpirationDate(expirationDate);
+
+		return cpConfigurationListPersistence.update(cpConfigurationList);
+	}
+
+	@Indexable(type = IndexableType.DELETE)
+	@Override
+	@SystemEvent(type = SystemEventConstants.TYPE_DELETE)
+	public CPConfigurationList deleteCPConfigurationList(
+			CPConfigurationList cpConfigurationList)
+		throws PortalException {
+
+		if (cpConfigurationList.isMasterCPConfigurationList()) {
+			throw new CPConfigurationListCannotDeleteException();
+		}
+
+		cpConfigurationListPersistence.remove(cpConfigurationList);
+
+		_cpConfigurationEntryLocalService.deleteCPConfigurationEntries(
+			cpConfigurationList.getCPConfigurationListId());
+
+		return cpConfigurationList;
+	}
+
+	@Override
+	public CPConfigurationList deleteCPConfigurationList(
+			long cpConfigurationListId)
+		throws PortalException {
+
+		CPConfigurationList cpConfigurationList =
+			cpConfigurationListPersistence.findByPrimaryKey(
+				cpConfigurationListId);
+
+		return cpConfigurationListLocalService.deleteCPConfigurationList(
+			cpConfigurationList);
+	}
+
+	@Override
+	public void deleteCPConfigurationLists(long companyId)
+		throws PortalException {
+
+		List<CPConfigurationList> cpConfigurationLists =
+			cpConfigurationListPersistence.findByCompanyId(companyId);
+
+		for (CPConfigurationList cpConfigurationList : cpConfigurationLists) {
+			cpConfigurationListLocalService.forceDeleteCPConfigurationList(
+				cpConfigurationList);
+		}
+	}
+
+	@Indexable(type = IndexableType.DELETE)
+	@Override
+	@SystemEvent(type = SystemEventConstants.TYPE_DELETE)
+	public CPConfigurationList forceDeleteCPConfigurationList(
+			CPConfigurationList cpConfigurationList)
+		throws PortalException {
+
+		cpConfigurationListPersistence.remove(cpConfigurationList);
+
+		_cpConfigurationEntryLocalService.deleteCPConfigurationEntries(
+			cpConfigurationList.getCPConfigurationListId());
+
+		return cpConfigurationList;
+	}
+
+	@Override
+	public List<CPConfigurationList> getCPConfigurationLists(
+		long groupId, long companyId) {
+
+		return cpConfigurationListPersistence.findByG_C(groupId, companyId);
+	}
+
+	@Override
+	public CPConfigurationList getMasterCPConfigurationList(long groupId)
+		throws NoSuchCPConfigurationListException {
+
+		return cpConfigurationListPersistence.findByG_MasterCPConfigurationList(
```

Comment:
Can we change the finder name to `G_M` instead of `G_MasterCPConfigurationList`?